src/main/java/forum/entity/Thread.java

```diff
@@ -0,0 +1,84 @@
+package forum.entity;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="thread")
+public class Thread {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id")
+	private int id;
+
+	@Column(name = "title")
+	private String title;
+	
+	@Column(name="content")
+	private String content;
+	
+	
+	@ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE,
```

Comment:
Can you clarify to me the intent of cascading here? Can you give me some example case?

Reply:
Cascading here makes it so that when the entity Thread goes through different states, the category for that thread will go though those states too. So if we would for example REFRESH (Reload/synch data from database) an instance of Thread object, it will also REFRESH the instance for an instance of Category object for it. Here we want to assign every cascade type except REMOVE so that if a thread is removed, it will NOT remove the category it belongs to.